hometask-2/src/data-access/user-repository.ts

```diff
@@ -0,0 +1,80 @@
+import { Op } from 'sequelize';
+import { v4 as uuidv4 } from 'uuid';
+
+import { User } from '../types';
+import { getUserModel } from './models/user';
+import { normalizeUser, normalizeUserRecord } from './normalizers/user';
+
+export async function getUserById(id: string): Promise<User | null> {
+    const user = await getUserModel().findOne({
+        where: { id, is_deleted: false },
+        raw: true
+    });
+
+    return normalizeUser(user);
+}
+
+export async function addUser(user: Omit<User, 'id | isDeleted'>): Promise<string> {
+    const generatedId = uuidv4();
+    const newUser = { ...user, id: generatedId, isDeleted: false };
```

Comment:
If you want to ideas of extra improvements, outside of homework, you can try to hash the user passwords.

Reply:
I'll try to do it in hometask about adding Authorization if I have time